packages/replay/src/coreHandlers/handleClick.ts

```diff
@@ -0,0 +1,302 @@
+    this._threshold = slowClickConfig.threshold / 1000;
+    this._scollTimeout = slowClickConfig.scrollTimeout / 1000;
+    this._replay = replay;
+    this._ignoreSelector = slowClickConfig.ignoreSelector;
+    this._addBreadcrumbEvent = _addBreadcrumbEvent;
+  }
+
+  /** Register click detection handlers on mutation or scroll. */
+  public addListeners(): void {
+    const mutationHandler = (): void => {
+      this._lastMutation = nowInSeconds();
+    };
+
+    const scrollHandler = (): void => {
+      this._lastScroll = nowInSeconds();
+    };
+
+    const clickHandler = (event: MouseEvent): void => {
+      if (!event.target) {
+        return;
+      }
+
+      const node = getClickTargetNode(event);
+      if (node) {
+        this._handleMultiClick(node as HTMLElement);
+      }
+    };
+
+    const obs = new MutationObserver(mutationHandler);
+
+    obs.observe(WINDOW.document.documentElement, {
+      attributes: true,
+      characterData: true,
+      childList: true,
+      subtree: true,
+    });
+
+    WINDOW.addEventListener('scroll', scrollHandler, { passive: true });
+    WINDOW.addEventListener('click', clickHandler, { passive: true });
+
+    this._teardown = () => {
+      WINDOW.removeEventListener('scroll', scrollHandler);
+      WINDOW.removeEventListener('click', clickHandler);
+
+      obs.disconnect();
+      this._clicks = [];
+      this._lastMutation = 0;
+      this._lastScroll = 0;
+    };
+  }
+
+  /** Clean up listeners. */
+  public removeListeners(): void {
+    if (this._teardown) {
+      this._teardown();
+    }
+
+    if (this._checkClickTimeout) {
+      clearTimeout(this._checkClickTimeout);
+    }
+  }
+
+  /** Handle a click */
+  public handleClick(breadcrumb: Breadcrumb, node: HTMLElement): void {
+    if (ignoreElement(node, this._ignoreSelector) || !isClickBreadcrumb(breadcrumb)) {
+      return;
+    }
+
+    const click = this._getClick(breadcrumb);
+
+    if (click) {
+      // this means a click on the same element was captured in the last 1s, so we consider this a multi click
+      return;
+    }
+
+    const newClick: Click = {
+      timestamp: breadcrumb.timestamp,
+      nodeId: breadcrumb.data.nodeId,
+      clickBreadcrumb: breadcrumb,
+      clickCount: 1,
+      node,
+    };
+    this._clicks.push(newClick);
+
+    // If this is the first new click, set a timeout to check for multi clicks
+    if (this._clicks.length === 1) {
+      this._scheduleCheckClicks();
+    }
+  }
+
+  /** Count multiple clicks on elements. */
+  private _handleMultiClick(node: HTMLElement): void {
+    const now = nowInSeconds();
+    const click = this._clicks.find(click => click.node === node && now - click.timestamp < this._multiClickTimeout);
+
+    if (!click) {
+      return;
+    }
+
+    // ignore VERY close timestamps - otherwise we record the initial timestamp twice!
+    if (click && Math.abs(click.timestamp - nowInSeconds()) > 0.01) {
```

Comment:
The problem is that we have two click listeners now. The one from the core breadcrumbs integration triggers the dom breadcrumb `ui.click` which is recorderd here via `handleClick()`. Then we have a separate click listener, which triggers right after this. So any click would trigger twice. Made more difficult by the timestamps being slightly off, as the logic we use in the core breadcrumbs integration is a bit more complex, so we can't really do an equal check in there 😬

Reply:
Can we separate the code paths so that the click from core breadcrumbs is handled outside and is only responsible for creating the `ui.click` breadcrumb?

Reply:
I've actually changed it up a bit. Completely separating it doesn't work easily as we then don't have access to the rrweb node stuff etc.

BUT I simply start out the `clickCount` as 0, then I can always increment it - that should work fine actually, I think, as the multi click handler should always run after the other handler.

Reply:
ok I'll take another look -- re: rrweb node, we should be able to get that information in the new handler. It'd be duplicating some work, but could be worth it if it simplifies logic.